src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/PhysicalAddress.cs

```diff
@@ -90,7 +90,7 @@ public override bool Equals(object? comparand)
 
         public override string ToString()
         {
-            return Convert.ToHexString(_address.AsSpan());
+            return HexConverter.ToString(_address.AsSpan(), HexConverter.Casing.Upper);
```

Comment:
It's concerning if the brand new public API is insufficient.

Reply:
I'm not concerted as we are talking about nanoseconds here.
A public API with input validation will never be a match for an internal method that gets straight to business. Especially with such short input values (16 bytes).

There is also #39702 to improve things (perf). The initial PR adding the public API was kept simple by design (to get it in for .NET 5).